packages/shared-ux/chrome/navigation/types/index.ts

```diff
@@ -40,93 +40,111 @@ export interface NavigationKibanaDependencies {
   };
 }
 
+/** @public */
+export type ChromeNavigationLink = string;
+
 /**
- * Props for the `NavItem` component representing the content of a navigational item with optional children.
+ * Chrome navigatioin node definition.
+ *
  * @public
  */
-export type NavItemProps<T = unknown> = Pick<EuiSideNavItemType<T>, 'id' | 'name'> & {
-  /**
-   * Nav Items
-   */
-  items?: Array<NavItemProps<T>>;
-  /**
-   * Href for a link destination
-   * Example: /app/fleet
-   */
-  href?: string;
-};
+export interface ChromeNavigationNode {
+  /** An optional id. If not provided a link must be passed */
+  id?: string;
+  /** An optional title for the node */
+  title?: string | ReactNode;
+  /** An optional eui icon */
+  icon?: string;
+  /** An app id or a deeplink id */
+  link?: ChromeNavigationLink;
+  /** Sub navigation item for this node */
+  items?: ChromeNavigationNode[];
+}
 
 /**
+ * Chrome navigation definition used internally in the components.
+ * Each "link" (if present) has been converted to a propert href. Additional metadata has been added
+ * like the "isActive" flag or the "path" (indicating the full path of the node in the nav tree).
+ *
  * @public
  */
-export interface PlatformSectionConfig {
-  enabled?: boolean;
-  properties?: Record<string, PlatformSectionConfig>;
+export interface ChromeNavigationNodeViewModel extends Omit<ChromeNavigationNode, 'items' | 'id'> {
+  id: string;
+  /**
+   * Full path that points to this node (includes all parent ids). If not set
+   * the path is the id
+   */
+  path?: string;
```

Comment:
For what do you think this will be used? 
Wouldn't `path: string[]` be more convenient? 
Will this path include this node as the last item? or only parents?

Reply:
This path includes everything. We could keep an array, but at the same time we can always build the array later if needed.

```ts
const path: string = 'stackManagement.someIntermediateSection.indexManagement.someDeepLinkId'
```